### DISCUSSION TITLE
Discovery between 2 PC's doesn't work

### DISCUSSION THREAD

**Discussion by Nir-Az** [eProsima/Fast-DDS#2612] [Q&A] [answered]
Hi,

I am tring to write a simple subscriber/publisher between 2 PC's 
Publisher = Windows
Subscriber = Linux
with FastDDS 2.6.0 and the discovery phase doesn't work.

I connected the 2 PC's  P2P with a direct Lan cable, and used default settings like the HelloWorld example.
I also tried setting a static IP for both
Publisher 100.0.0.1
Subscriber 100.0.0.2

But it's not working.

Also added 100.0.0.2 as `initial_peer` for the publisher.. nothing

Any good way to debug this kind of issues? I see Publisher -> subscriber ICMP messages are getting a port unreachable error.
Wireshark dump from the publisher PC [here](https://easyupload.io/8o1jtb).

Appreciate any help understanding the issues..

**Comment by MiguelCompany** [comment, marked as answer]
@Nir-Az I took a look at your capture, and it looks like the Publisher is deaf. Maybe the Windows firewall is active..

As for the ICMP messages, using an `initial_peer` with port 0 will result in trying to communicate on the default port for participants 0 to 3, that is why there are 3 port unreachable messages every time the publisher sends 4 DATA(p) announcements, as only the one directed to the port for participant 1 (port 7412) succeeds.

The scenario you mention is one we test before every Fast DDS release. We have just checked it here, and it works as expected.

**Reply by Nir-Az** [reply]
@MiguelCompany  Thanks for the detailed explanation.
I will try to see how to overcome the firewall issues.
